### PR TITLE
[Doc] Clarify the contributor guide

### LINKED ISSUE #41402: [Doc] Contributor guide makes no mention of forking the repo
To contribute to Ray, the user first needs to fork the repository.  If they try to directly push a branch to the main repo instead of their fork, they will encounter a 403 permission denied error.

However, "fork" isn't mentioned anywhere in the contributor guide https://docs.ray.io/en/latest/ray-contribute/getting-involved.html.  Right now it's only mentioned in a separate page linked from one of the steps https://docs.ray.io/en/latest/ray-contribute/development.html#clone-the-repository, and even there it's a bit buried.

We should make it clear that the first step is to fork the Ray repository.

_Originally posted by @architkulkarni in https://github.com/ray-project/ray/issues/41277#issuecomment-1828352187_

### PR DESCRIPTION
**Pull request by justina777**
## Why are these changes needed?
To clarify the first step in the contribution guide is to fork the repository before checkout. 

## Related issue number
Close #41402 

## Checks

- [v] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [v] I've run `scripts/format.sh` to lint the changes in this PR.
- [v] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Thanks for the contribution!

**Comment by architkulkarni** [comment]
@angelinalg could you please give codeowner approval to this PR?

File: doc/source/ray-contribute/getting-involved.rst [hunk 1]
```diff
@@ -51,7 +51,7 @@ ones labeled `"good first issue" <https://github.com/ray-project/ray/issues?utf8
 Setting up your development environment
 ---------------------------------------
 
-To edit the Ray source code, you'll want to checkout the repository and also build Ray from source. Follow :ref:`these instructions for building <building-ray>` a local copy of Ray to easily make changes.
+To edit the Ray source code, you'll want to fork the repository, clone it and build Ray from source. Follow :ref:`these instructions for building <building-ray>` a local copy of Ray to easily make changes.
```
Comment (angelinalg) [inline doc/source/ray-contribute/getting-involved.rst:54]:
```suggestion
To edit the Ray source code, fork the repository, clone it, and build Ray from source. Follow :ref:`these instructions for building <building-ray>` a local copy of Ray to easily make changes.
```

**Review by angelinalg — APPROVED** [review]
Thanks for submitting this docs improvement! Just a little style nit, if you wouldn't mind considering.
Thanks!

File: doc/source/ray-contribute/getting-involved.rst (hunk 1 above)
Comment (justina777) [reply doc/source/ray-contribute/getting-involved.rst:54]: Thanks for the suggestion. will the update it

**Comment by dmatrix** [comment]
Congrats @justina777 to have merged your first PR!

**Pull request merged**